modules/shadertools/test/lib/assemble-shaders.spec.js

```diff
@@ -115,3 +121,70 @@ test('assembleShaders#getUniforms', t => {
 
   t.end();
 });
+
+test('assembleShaders#shaderhooks', t => {
```

Comment:
Just an idea: you can also use `compileVertexShader` and `compileFragmentShader` to transpile the assembled shaders and test their output.

Reply:
Oh nice! I'll give that a shot.